CondTools/Ecal/plugins/StoreESCondition.cc

```diff
@@ -0,0 +1,449 @@
+  strcpy(header,asctime(timeinfo));
+  strcpy(header,user);
+}
+
+ESThresholds* StoreESCondition::readESThresholdsFromFile(const char* inputFile) {
+
+  std::ifstream ESThresholdsFile(edm::FileInPath(inputFile).fullPath().c_str());
+  float ts2, zs;   //2nd time sample, ZS threshold
+  ESThresholdsFile >> ts2;
+  ESThresholdsFile >> zs;
+  ESThresholds* esThresholds = new ESThresholds(ts2, zs);
+
+  return esThresholds;
+}
+
+ESEEIntercalibConstants* StoreESCondition::readESEEIntercalibConstantsFromFile(const char* inputFile) {
+
+  std::ifstream ESEEIntercalibFile(edm::FileInPath(inputFile).fullPath().c_str());
+  float gammaLow0, alphaLow0, gammaHigh0, alphaHigh0, gammaLow1, alphaLow1, gammaHigh1, alphaHigh1,
+    gammaLow2, alphaLow2, gammaHigh2, alphaHigh2, gammaLow3, alphaLow3, gammaHigh3, alphaHigh3;
+  //  const float ESEEIntercalibValue[16];
+  //  for (int i = 0; i < 16; ++i) {
+  //    ESEEIntercalibFile >> ESEEIntercalibValue[i];
+  //  }
+  //  ESEEIntercalibConstants* eseeIntercalibConstants = new ESEEIntercalibConstants(ESEEIntercalibValue);
+  ESEEIntercalibFile >> gammaLow0;
+  ESEEIntercalibFile >> alphaLow0;
+  ESEEIntercalibFile >> gammaHigh0;
+  ESEEIntercalibFile >> alphaHigh0;
+  ESEEIntercalibFile >> gammaLow1;
+  ESEEIntercalibFile >> alphaLow1;
+  ESEEIntercalibFile >> gammaHigh1;
+  ESEEIntercalibFile >> alphaHigh1;
+  ESEEIntercalibFile >> gammaLow2;
+  ESEEIntercalibFile >> alphaLow2;
+  ESEEIntercalibFile >> gammaHigh2;
+  ESEEIntercalibFile >> alphaHigh2;
+  ESEEIntercalibFile >> gammaLow3;
+  ESEEIntercalibFile >> alphaLow3;
+  ESEEIntercalibFile >> gammaHigh3;
+  ESEEIntercalibFile >> alphaHigh3;
+  ESEEIntercalibConstants* eseeIntercalibConstants = new ESEEIntercalibConstants(gammaLow0, alphaLow0, gammaHigh0, alphaHigh0, 
+										 gammaLow1, alphaLow1, gammaHigh1, alphaHigh1,
+										 gammaLow2, alphaLow2, gammaHigh2, alphaHigh2, 
+										 gammaLow3, alphaLow3, gammaHigh3, alphaHigh3);
+
+  return eseeIntercalibConstants;
+
+}
+
+ESMissingEnergyCalibration* StoreESCondition::readESMissingEnergyFromFile(const char* inputFile) {
+
+  std::ifstream ESMissingEnergyFile(edm::FileInPath(inputFile).fullPath().c_str());
+  float ConstAEta0, ConstBEta0, ConstAEta1, ConstBEta1, ConstAEta2, ConstBEta2, ConstAEta3, ConstBEta3;
+  ESMissingEnergyFile >> ConstAEta0;
+  ESMissingEnergyFile >> ConstBEta0;
+  ESMissingEnergyFile >> ConstAEta1;
+  ESMissingEnergyFile >> ConstBEta1;
+  ESMissingEnergyFile >> ConstAEta2;
+  ESMissingEnergyFile >> ConstBEta2;
+  ESMissingEnergyFile >> ConstAEta3;
+  ESMissingEnergyFile >> ConstBEta3;
+  ESMissingEnergyCalibration* esMissingEnergy = new ESMissingEnergyCalibration(ConstAEta0, ConstBEta0, ConstAEta1, ConstBEta1, 
+									       ConstAEta2, ConstBEta2, ConstAEta3, ConstBEta3); 
+
+  return esMissingEnergy;
+}
+
+ESPedestals* StoreESCondition::readESPedestalsFromFile(const char* inputFile) {
+
+  ESPedestals* esPedestals = new ESPedestals();
+
+  int ped[2][2][40][40][32];
```

Comment:
this can be initialized via 
int ped..... ={}

second - most of these magic numbers appear to have an constant defining them (used below) - please use those constants instead of the magic numbers

finally - is this needed at all - could you read the file and fill the pedestal object directly (maybe we already had this discussion?)